gtfs_rt_validator_api.py

```diff
@@ -25,42 +72,169 @@ def validate(gtfs_file, rt_path, out_file=None, verbose=False):
     stderr = subprocess.DEVNULL if not verbose else None
     stdout = subprocess.DEVNULL if not verbose else None
 
-    with TemporaryDirectory() as tmp_out_dir:
-        subprocess.check_call([
-            "java",
-            "-jar", JAR_PATH,
-            "-gtfs", gtfs_file,
-            "-gtfsRealtimePath", gtfs_file,
-            ], stderr=stderr, stdout=stdout)
+    subprocess.check_call([
+        "java",
+        "-jar", JAR_PATH,
+        "-gtfs", gtfs_file,
+        "-gtfsRealtimePath", rt_path,
+        "-sort", "name",
+        ], stderr=stderr, stdout=stdout)
 
 
 def validate_gcs_bucket(
-        project_id, token, gtfs_schedule_path, gtfs_rt_path,
-        gtfs_rt_glob="2021-08-01",
-        out_file=None, verbose=False
+        project_id, token, gtfs_schedule_path, gtfs_rt_glob_path=None,
+        out_dir=None, results_bucket=None, verbose=False, 
         ):
+    """
+    Fetch and validate GTFS RT data held in a google cloud bucket.
+
+    Parameters:
+        project_id: name of google cloud project.
+        token: token argument passed to gcsfs.GCSFileSystem.
+        gtfs_schedule_path: path to a folder holding unpacked GTFS schedule data.
+        gtfs_rt_glob_path: path that GCSFileSystem.glob can uses to list all RT files.
+            Note that this is assumed to have the form {datetime}/{itp_id}/{url_number}/filename.
+        out_dir: a directory to store fetched files and results in.
+        results_dir: a bucket path to copy results to.
```

Comment:
Doc param name doesn't match function name, see recommended change:

```suggestion
        results_bucket: a bucket path to copy results to.
```

Reply:
should be updated